lib/framedivider.rb

```diff
@@ -107,10 +167,10 @@ def opening(model, id, t)
   pts[:C] = {}
   pts[:D] = {} if four
 
-  pts[:A][:pt] = points[0]
-  pts[:B][:pt] = points[1]
-  pts[:C][:pt] = points[2]
-  pts[:D][:pt] = points[3] if four
+  pts[:A][:pt] = Topolys::Point3D.new(points[0].x, points[0].y, points[0].z)
+  pts[:B][:pt] = Topolys::Point3D.new(points[1].x, points[1].y, points[1].z)
+  pts[:C][:pt] = Topolys::Point3D.new(points[2].x, points[2].y, points[2].z)
+  pts[:D][:pt] = Topolys::Point3D.new(points[3].x, points[3].y, points[3].z) if four
```

Comment:
I state earlier that these functions are entirely OpenStudio - I lied. The new _rough opening_ vertices are estimated here from Topolys vector manipulations, then reconverted back to OpenStudio 3D points. In due course, I can recode to ensure a 100% OpenStudio solution.